### ISSUE TITLE
Itemducs cause Intense Server Lag when Target Inventories are Full

### ISSUE THREAD

**Issue by leumasme** [Darkosto/SkyFactory4Issues#320]

**Comment by Stqlth** [comment]
This, exactly the same for us, bonsai pots into chest/cabinets or cobblestone generators into dispensers or Automated users that are full F***s our tps on a 8b 2 player server.

**Comment by Rahmiel** [comment]
I was about to blame bonsai trees for this. I can confirm I am having a similar issue. Has anyone tested any other methods of transport other than itemducts?

**Comment by ktole1999** [comment]
ducts wont push if inventory is not available otherwise the servo jams. i had 4 banks of 60 crates using ducts while connected to 23 hopping bonsai.  even when crates were full my tps never dropped.

**Comment by Rahmiel** [comment]
> ducts wont push if inventory is not available otherwise the servo jams. i had 4 banks of 60 crates using ducts while connected to 23 hopping bonsai. even when crates were full my tps never dropped.

No, but hopping bonsai trees WILL push items, if they can't they will void the items. Something in this combination of things is causing extreme tps lag.

**Comment by Rahmiel** [comment]
This is a really odd interaction overall. The Bonsai Trees create almost no TPS lag on their own, just voiding all their items being attached to no inventory or even chests. However as soon as I connect them to item ducts, I go from 16ms MTT to 60+ms. This is not good at all.

I also tested Cyclic Item Cables, they don't cause the tps lag, however they don't seem to work with Void upgrades or Shipping Containers.

**Comment by ktole1999** [comment]
are the bonsai in close proximity causing duct loops?

**Comment by Rahmiel** [comment]
Just an update for those waiting for some kind of progression or something you can do to work around this. Bonsai trees on ME Interfaces with Storage Interfaces on Shipping Bins (or whatever you use) and a Last Resort (Set Priority to -1000 or less) Storage Interface on a Trash Can will work perfectly. Minimal hit to MTT (+4MS overall with 25 bonsai trees going compared to +30MS overall with 20 Bonsai Trees connected to Itemducts).
